src/main/scala/com/databricks/labs/overwatch/pipeline/Snapshot.scala

```diff
@@ -0,0 +1,376 @@
+                                          excludes: Option[String] = Some("")
+                                        ): Unit = {
+
+
+    val sourceToSnapFiltered = filteredTables(pipelineTables,excludes)
+
+    val cloneSpecs = buildCloneSpecs("DEEP",sourceToSnapFiltered)
+    snapStream(cloneSpecs)
+  }
+
+  private[overwatch] def snap(
+                               pipelineTables : Array[PipelineTable],
+                               cloneLevel: String = "DEEP",
+                               excludes: Option[String] = Some("")
+                             ): Unit= {
+    val acceptableCloneLevels = Array("DEEP", "SHALLOW")
+    require(acceptableCloneLevels.contains(cloneLevel.toUpperCase), s"SNAP CLONE ERROR: cloneLevel provided is " +
+      s"$cloneLevel. CloneLevels supported are ${acceptableCloneLevels.mkString(",")}.")
+
+    val sourceToSnapFiltered = filteredTables(pipelineTables,excludes)
+
+    val cloneSpecs = buildCloneSpecs(cloneLevel,sourceToSnapFiltered)
+    val cloneReport = Helpers.parClone(cloneSpecs)
+    val cloneReportPath = s"${snapshotRootPath}/clone_report/"
+    cloneReport.toDS.write.format("delta").mode("append").save(cloneReportPath)
+    logger.log(Level.INFO, s"Clone report has been generated to this path: ${cloneReportPath}")
+  }
+
+}
+
+object Snapshot extends SparkSessionWrapper {
+
+  private val logger: Logger = Logger.getLogger(this.getClass)
+
+  def isValid(sourceETLDB : String,
+              snapshotType : String,
+              pipeline : String = "Bronze,Silver,Gold",
+              cloneLevel: String = "DEEP",
+              processType : String = "Snapshot"
+             ): Boolean = {
+
+    // Check whether sourceETLDB is Overwatch Database
+    if (processType == "Snapshot") {
+      val isOverwatchDB = spark.sessionState.catalog.getDatabaseMetadata(sourceETLDB).properties.getOrElse("OVERWATCHDB", "FALSE").toBoolean
+      if (isOverwatchDB) {
+        println(s"${sourceETLDB} is Overwatch Database and suitable for Snapshot")
+      } else {
+        val errMsg = s"${sourceETLDB} is Not Overwatch Database and not suitable for Snapshot"
+        throw new BadConfigException(errMsg)
+      }
+
+      // Snapshot Type should be Incremental or Full
+      if (snapshotType.toLowerCase() == "incremental" || snapshotType.toLowerCase() == "full") {
+        println(s"Snapshot Type is Suitable for Snapshot Process. Provided SnapshotType value is ${snapshotType}")
+      } else {
+        val errMsg = s"Provided SnapshotType value is ${snapshotType}. SnapshotType value should be either Full or Incremental. Can Not Proceed with Snapshot"
+        throw new BadConfigException(errMsg)
+      }
+
+      // Pipeline Should be Bronze, Sliver Or Gold
+
+      val pipelineList = pipeline.split(",").map(_.trim)
+
+      for (layer <- pipelineList) {
+        if (layer.toLowerCase() == "bronze" || layer.toLowerCase()  == "silver" || layer.toLowerCase()  == "gold") {
```

Comment:
did you forget to turn the positive (layer === X) into the negative (layer =!= X)?

Reply:
Done